utils/zerovec/src/map/map.rs

```diff
@@ -98,6 +98,12 @@ where
             values: V::Container::zvl_new(),
         }
     }
+
+    #[doc(hidden)] // Crabbake internal
+    pub fn from_parts_unchecked(keys: K::Container, values: V::Container) -> Self {
```

Comment:
Discussion: are this and similar constructors better as doc-hidden, doc-hidden unsafe, or pub unsafe?

Reply:
I don't think they should be doc-hidden, we shouldn't be special casing CrabBake too much. This one doesn't _need_ to be unsafe but I'm not against making it unsafe anyway.

Reply:
I made this doc-hidden because it exposes implementation details. The "parts" it refers to might change in the future.

Reply:
oh, that's a fair point i guess. It's unlikely, but makes sense

Reply:
How about between safe hidden and unsafe hidden?

I lean toward making these internal constructors unsafe because (1) there are invariants that need to be upheld, (2) people can't accidentally use the functions in an unsafe way, and (3) I see no compelling reason that the methods should *not* be unsafe

Reply:
Oh whoops I meant to make them doc-hidden unsafe.